### PR TITLE
Enable per camera control of optimization of different intrinsics components

### PR DESCRIPTION
**Pull request by ghost**
Add a lock for scale, offset, distortion

This lock is considered during bundle

### CODE REVIEW

File: src/aliceVision/sfm/BundleAdjustmentSymbolicCeres.cpp [hunk 1]
```diff
@@ -574,20 +599,22 @@ void BundleAdjustmentSymbolicCeres::addIntrinsicsToProblem(const sfmData::SfMDat
...
     double* intrinsicBlockPtr = intrinsicBlock.data();
+
     problem.AddParameterBlock(intrinsicBlockPtr, intrinsicBlock.size());
 
     // add intrinsic parameter to the all parameters blocks pointers list
     _allParametersBlocks.push_back(intrinsicBlockPtr);
 
     // keep the camera intrinsic constant
-    if(intrinsicPtr->isLocked() || !refineIntrinsics || getIntrinsicState(intrinsicId) == EParameterState::CONSTANT)
+    if(intrinsicPtr->isLocked() || getIntrinsicState(intrinsicId) == EParameterState::CONSTANT)
```
Comment (fabiencastan) [inline src/aliceVision/sfm/BundleAdjustmentSymbolicCeres.cpp:609]: Why did you remove "!refineIntrinsics"?

File: src/aliceVision/sfm/BundleAdjustmentCeres.cpp [hunk 2]
```diff
@@ -619,14 +644,15 @@ void BundleAdjustmentCeres::addIntrinsicsToProblem(const sfmData::SfMData& sfmDa
     _allParametersBlocks.push_back(intrinsicBlockPtr);
 
     // keep the camera intrinsic constant
-    if(intrinsicPtr->isLocked() || !refineIntrinsics || getIntrinsicState(intrinsicId) == EParameterState::CONSTANT)
+    if(intrinsicPtr->isLocked() || getIntrinsicState(intrinsicId) == EParameterState::CONSTANT)
```
Comment (fabiencastan) [inline src/aliceVision/sfm/BundleAdjustmentCeres.cpp:647]: same here

Comment (servantf) [reply src/aliceVision/sfm/BundleAdjustmentCeres.cpp:647]: Erreur, tu peux corriger stp ?

**Comment by fabiencastan** [comment]
Integrated in https://github.com/alicevision/AliceVision/pull/1157
